client/components/data/query-site-comments/index.jsx

```diff
@@ -18,8 +18,8 @@ export class QuerySiteComments extends Component {
 		requestComments( this.props );
 	}
 
-	componentDidUpdate( { siteId } ) {
-		if ( siteId !== this.props.siteId ) {
+	componentDidUpdate( { siteId, status } ) {
+		if ( siteId !== this.props.siteId || status !== this.props.status ) {
 			requestComments( this.props );
```

Comment:
Not sure I get what you mean, but in the `QuerySiteComments` case, we already had the `status` prop around (used here: https://github.com/Automattic/wp-calypso/pull/16082/files#diff-356c28fb0b0ef5e9073f6233277d14e4L269).